lua/starfall/libs_sv/prop.lua

```diff
@@ -430,9 +458,57 @@ function props_library.createSent(pos, ang, class, frozen)
 		end
 
 		hookcall = "PlayerSpawnedVehicle"
-
+	elseif sent2 then
+		if ply ~= NULL then
+			if (scripted_ents.GetStored(class).t.AdminOnly and not ply:IsAdmin()) then SF.Throw("This sent is admin only!", 2) end
+			if gamemode.Call("PlayerSpawnSENT", ply, class) == false then SF.Throw("Another hook prevented the sent from spawning", 2) end
+		end
+		
+		local enttbl = {}
+		data = data or {}
+		
+		-- Apply data
+		for param, org in pairs(sent2[1]) do
+			local value = data[param]
+			
+			if value then
+				value = ounwrap(value) or value
+				
+				checkluatype(value, org[2], nil, "Parameter: " .. param)
+				
+				enttbl[org[1]] = value
```

Comment:
We need to check if TYPE_TABLE and create a new table and copy the specific indeces if so. Users' tables can't be trusted.

Reply:
It would also be good to detect invalid keys in 'data' and throw an error.